### PR TITLE
refactor(simple_planning_simulator): rework parameters

### PR DESCRIPTION
**Pull request by PhoebeWu21**
## Description
Implement the ROS Node configuration layout described in https://github.com/orgs/autowarefoundation/discussions/3371 for the `simple_planning_simulator` package.

- Remove the default value from the source code in order to ensure all parameter values are passed from the parameter files.
- Create the schema

## Tests performed
Package built and launch locally.
`colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release --packages-up-to simple_planning_simulator`

## Effects on system behavior
More reliable and faster parameter configuration file creation.

## Pre-review checklist for the PR author
The PR author **must** check the checkboxes below when creating the PR.
- [x] I've confirmed the [contribution guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/).
- [x] The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).

## In-review checklist for the PR reviewers
The PR reviewers **must** check the checkboxes below before approval.
- [ ] The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author
The PR author **must** check the checkboxes below before merging.
- [x] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

**Comment by ambroise-arm** [comment]
The json-schema CI check fails because the config file should be under `../config/*.param.yaml`. Could you move it there and ensure that if there are other packages making use of the config file they are updated to the new path as well?

**Comment by PhoebeWu21** [comment]
> The json-schema CI check fails because the config file should be under `../config/*.param.yaml`. Could you move it there and ensure that if there are other packages making use of the config file they are updated to the new path as well?

@ambroise-arm Got it! Done.

**Comment by PhoebeWu21** [comment]
> The json-schema CI check fails because the config file should be under `../config/*.param.yaml`. Could you move it there and ensure that if there are other packages making use of the config file they are updated to the new path as well?

@ambroise-arm Got it! Done.

File: simulator/simple_planning_simulator/config/vehicle_characteristics.param.yaml
Comment (ambroise-arm) [inline simulator/simple_planning_simulator/config/vehicle_characteristics.param.yaml:1]: This one fails the check in CI. Because it doesn't match against simple_planning_simulator.schema.json (of course it doesn't, it is not meant to).
If it was needed because this packages had 2 nodes each needing a config file, we would need a second json schema file and a modification in the CI files (already needed for another PR). But here I just don't see how this config file is used. Could it just be removed? (the file and the its use in the launch file)

Comment (PhoebeWu21) [reply simulator/simple_planning_simulator/config/vehicle_characteristics.param.yaml:1]: Got it. Thank you for the explanation. 
Already fix it.

**Comment by satoshi-ota** [comment]
@PhoebeWu21 Following error has occured by this PR. Could you confirm whether the autoware can be launch correctly with your branch?

```
[ERROR] [simple_planning_simulator_exe-53]: process has died [pid 572727, exit code -6, cmd '/home/satoshi/pilot-auto/install/simple_planning_simulator/lib/simple_planning_simulator/simple_planning_simulator_exe --ros-args -r __node:=simple_planning_simulator -r __ns:=/simulation -p use_sim_time:=False -p wheel_radius:=0.383 -p wheel_width:=0.235 -p wheel_base:=2.79 -p wheel_tread:=1.64 -p front_overhang:=1.0 -p 
rear_overhang:=1.1 -p left_overhang:=0.128 -p right_overhang:=0.128 -p vehicle_height:=2.5 -p max_steer_angle:=0.7 --params-file /tmp/launch_params_cm1fjop5 --params-file /tmp/launch_params_99funsy_ -r input/vector_map:=/map/vector_map -r input/initialpose:=/initialpose3d -r input/ackermann_control_command:=/control/command/control_cmd -r input/manual_ackermann_control_command:=/vehicle/command/manual_control_cmd
 -r input/gear_command:=/control/command/gear_cmd -r input/manual_gear_command:=/vehicle/command/manual_gear_command -r input/turn_indicators_command:=/control/command/turn_indicators_cmd -r input/hazard_lights_command:=/control/command/hazard_lights_cmd -r input/trajectory:=/planning/scenario_planning/trajectory -r input/engage:=/vehicle/engage -r input/control_mode_request:=/control/control_mode_request -r outp
ut/twist:=/vehicle/status/velocity_status -r output/odometry:=/localization/kinematic_state -r output/acceleration:=/localization/acceleration -r output/imu:=/sensing/imu/imu_data -r output/steering:=/vehicle/status/steering_status -r output/gear_report:=/vehicle/status/gear_status -r output/turn_indicators_report:=/vehicle/status/turn_indicators_status -r output/hazard_lights_report:=/vehicle/status/hazard_light
s_status -r output/control_mode_report:=/vehicle/status/control_mode'].                                                                                                                                                                                                                                                                                                                                                         
[simple_planning_simulator_exe-53] terminate called after throwing an instance of 'rclcpp::exceptions::UninitializedStaticallyTypedParameterException'                                                                                                                                                                                                                                                                          
[simple_planning_simulator_exe-53]   what():  Statically typed parameter 'initial_engage_state' must be initialized.
```

**Comment by PhoebeWu21** [comment]
> @PhoebeWu21 Following error has occured by this PR. Could you confirm whether the autoware can be launch correctly with your branch?
> 
> ```
> [ERROR] [simple_planning_simulator_exe-53]: process has died [pid 572727, exit code -6, cmd '/home/satoshi/pilot-auto/install/simple_planning_simulator/lib/simple_planning_simulator/simple_planning_simulator_exe --ros-args -r __node:=simple_planning_simulator -r __ns:=/simulation -p use_sim_time:=False -p wheel_radius:=0.383 -p wheel_width:=0.235 -p wheel_base:=2.79 -p wheel_tread:=1.64 -p front_overhang:=1.0 -p 
> rear_overhang:=1.1 -p left_overhang:=0.128 -p right_overhang:=0.128 -p vehicle_height:=2.5 -p max_steer_angle:=0.7 --params-file /tmp/launch_params_cm1fjop5 --params-file /tmp/launch_params_99funsy_ -r input/vector_map:=/map/vector_map -r input/initialpose:=/initialpose3d -r input/ackermann_control_command:=/control/command/control_cmd -r input/manual_ackermann_control_command:=/vehicle/command/manual_control_cmd
>  -r input/gear_command:=/control/command/gear_cmd -r input/manual_gear_command:=/vehicle/command/manual_gear_command -r input/turn_indicators_command:=/control/command/turn_indicators_cmd -r input/hazard_lights_command:=/control/command/hazard_lights_cmd -r input/trajectory:=/planning/scenario_planning/trajectory -r input/engage:=/vehicle/engage -r input/control_mode_request:=/control/control_mode_request -r outp
> ut/twist:=/vehicle/status/velocity_status -r output/odometry:=/localization/kinematic_state -r output/acceleration:=/localization/acceleration -r output/imu:=/sensing/imu/imu_data -r output/steering:=/vehicle/status/steering_status -r output/gear_report:=/vehicle/status/gear_status -r output/turn_indicators_report:=/vehicle/status/turn_indicators_status -r output/hazard_lights_report:=/vehicle/status/hazard_light
> s_status -r output/control_mode_report:=/vehicle/status/control_mode'].                                                                                                                                                                                                                                                                                                                                                         
> [simple_planning_simulator_exe-53] terminate called after throwing an instance of 'rclcpp::exceptions::UninitializedStaticallyTypedParameterException'                                                                                                                                                                                                                                                                          
> [simple_planning_simulator_exe-53]   what():  Statically typed parameter 'initial_engage_state' must be initialized.
> ```

Hi @satoshi-ota, I did successfully launch in local.
![image](https://github.com/autowarefoundation/autoware.universe/assets/36169304/31f1380f-870f-453a-94a7-a28f9e39bb69)
I have no idea what happened. I will recheck detailly again.

**Comment by satoshi-ota** [comment]
Hi @PhoebeWu21 Thanks for your investigation.
Hmm... OK. I'll recheck, too.

**Comment by satoshi-ota** [comment]
@PhoebeWu21 I guess you can see above errors if you try following tutorials.
https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/planning-simulation/#lane-driving-scenario

**Comment by satoshi-ota** [comment]
When it launches planning_simulator from `autoware_launch` repos, the launch file includes [other](https://github.com/autowarefoundation/sample_vehicle_launch/blob/627068935b12ec6d6121e6a1b885e31d564b04c5/sample_vehicle_description/config/simulator_model.param.yaml#L1-L18) sim model yaml [here](https://github.com/autowarefoundation/autoware.universe/blob/b9e98f1e43f350513cffd347a0f3c9d9a692e89e/launch/tier4_simulator_launch/launch/simulator.launch.xml#L139).

```
ros2 launch autoware_launch planning_simulator.launch.xml map_path:=$HOME/autoware_map/sample-map-planning vehicle_model:=sample_vehicle sensor_model:=sample_sensor_kit
```

And, it seems that `initial_engage_state` param doesn't exist in the yaml file. I'm sure that this is the cause of the error.

**Comment by PhoebeWu21** [comment]
@satoshi-ota  I see, thank you!

**Comment by PhoebeWu21** [comment]
> I add back `{ "initial_engage_state": LaunchConfiguration("initial_engage_state"), },` in simple_planning_simulator.launch.py file. It seems work for launching planning_simulator. But the error `Caught exception in launch (see debug for traceback): launch configuration 'initial_engage_state' does not exist` will occur while launching the single simple_planning_simulator node.

@satoshi-ota Hi, I am not sure how to deal with this situation, since I did not make any other change to the launch file.

**Comment by satoshi-ota** [comment]
@PhoebeWu21 Sorry for late reply. I'll investigate that error.

**Comment by satoshi-ota** [comment]
> When it launches planning_simulator from `autoware_launch` repos, the launch file includes [other](https://github.com/autowarefoundation/sample_vehicle_launch/blob/627068935b12ec6d6121e6a1b885e31d564b04c5/sample_vehicle_description/config/simulator_model.param.yaml#L1-L18) sim model yaml [here](https://github.com/autowarefoundation/autoware.universe/blob/b9e98f1e43f350513cffd347a0f3c9d9a692e89e/launch/tier4_simulator_launch/launch/simulator.launch.xml#L139).
> 
> ```
> ros2 launch autoware_launch planning_simulator.launch.xml map_path:=$HOME/autoware_map/sample-map-planning vehicle_model:=sample_vehicle sensor_model:=sample_sensor_kit
> ```
> 
> And, it seems that `initial_engage_state` param doesn't exist in the yaml file. I'm sure that this is the cause of the error.

Hi @PhoebeWu21 
I think you need to add **NOT ONLY** `initial_engage_state` but also `enable_road_slope_simulation`, `pos_noise_stddev`, etc...

And, as I wrote, when we call planning_simulator.launch.xml, **[this](https://github.com/autowarefoundation/sample_vehicle_launch/blob/627068935b12ec6d6121e6a1b885e31d564b04c5/sample_vehicle_description/config/simulator_model.param.yaml#L1-L18)** file will be loaded. Actually, I confirmed that I could launch Psim with your branch after I added missing params to **[this](https://github.com/autowarefoundation/sample_vehicle_launch/blob/627068935b12ec6d6121e6a1b885e31d564b04c5/sample_vehicle_description/config/simulator_model.param.yaml#L1-L18)** file.

In short, we have to add missing params to `simulator_model.param.yaml` of **ALL** vehicle description packages (TIER IV has a lot of vehicle descriptions...) before removing default values in cpp file at first. So, I'll ask TIER IV internal member to help adding missing params.

**Comment by PhoebeWu21** [comment]
Hi @satoshi-ota,
I see. So I think I don't need do modify my branch until the missing params in simulator_model.param.yaml be added, right?

**Comment by idorobotics** [comment]
> In short, we have to add missing params to simulator_model.param.yaml of ALL vehicle description packages (TIER IV has a lot of vehicle descriptions...) before removing default values in cpp file at first. So, I'll ask TIER IV internal member to help adding missing params.

@satoshi-ota - have you received feedback on adding the missing params? Can @PhoebeWu21 continue with this task or will you be taking over to complete it?

**Comment by idorobotics** [comment]
@satoshi-ota could you update the current status of this PR?

**Comment by mitsudome-r** [comment]
@mitsudome-r To resolve the conflicts and do tests with launch files.

**Comment by mitsudome-r** [comment]
I have rebased the branch to the latest main and resolved conflicts.
There seem to be some parameters added after the PR was made so those need to be added to schema file as well.

**Comment by mitsudome-r** [comment]
I will test locally if the launch file functions correctly.
